src/context/WalletProvider/NativeWallet/components/NativeLegacySuccess.tsx

```diff
@@ -1,16 +1,19 @@
-import type { Vault } from '@shapeshiftoss/hdwallet-native-vault'
 import { useCallback } from 'react'
 import { useHistory, useLocation } from 'react-router-dom'
 import { NativeWalletRoutes } from 'context/WalletProvider/types'
 
+import type { LocationState } from '../types'
 import { LegacyLoginSuccess } from './LegacyMigration/LegacyLoginSuccess'
 
 export const NativeLegacySuccess = () => {
-  const history = useHistory<{ vault: Vault }>()
-  const location = useLocation<{ vault: Vault }>()
+  const history = useHistory<LocationState>()
+  const location = useLocation<LocationState>()
 
   const handleContinue = useCallback(() => {
-    history.push(NativeWalletRoutes.Create, location.state)
+    history.push(NativeWalletRoutes.Create, {
+      vault: location.state.vault,
+      isLegacyWallet: true,
```

Comment:
Explicitly passing `isLegacyWallet` instead of previously assuming having `vault` defined is legacy wallet, as now by clicking back we pass back the `vault` also

Reply:
Remove me, see above

Reply:
Done in [d8091b3](https://github.com/shapeshift/web/pull/8885/commits/d8091b329778cd99a575de52a02ddf6598afd3ed)